carla/data/catalog/catalog.py

```diff
@@ -21,6 +21,10 @@ def __init__(self, data_name: str):
         self.name = data_name
         self.catalog: Dict[str, Any] = load_catalog("data_catalog.yaml", data_name)
 
+        for key in ["continous", "categorical", "immutable"]:
```

Comment:
That could also go in the load model function, so other consumers could benefit from it. Plus we can make it more generic and also add the mandatory keys if they were missing in the yaml

Reply:
I think the yaml is only used for the catalog models and datasets we provide, and a user should not manipulate or add points to it. An implementation of an UserDataset or UserModel would mean to inherit Data or MLModel and implement own logic to these properties.

WDYT

Reply:
True, but in this case were only handleing the catalog datasets anyhow right? I am saying that "making sure that `catalog` has all its keys and values we expect would be nicer abstracted away in the `load_catalog` function, so whenever we call this function we know all the things are there

Reply:
Ok, now I know what you mean, sorry😅 Makes totally sense! I'd put it inside the constructor of the catalog classes because I'm using the `load_catalog` method for MLModels and Data.